src/ejabberd_sm.erl

```diff
@@ -871,18 +871,19 @@ is_offline(#jid{luser = LUser, lserver = LServer}) ->
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
 %% @doc On new session, check if some existing connections need to be replace
--spec check_for_sessions_to_replace(HostType, JID) -> ReplacedPids when
+-spec check_for_sessions_to_replace(HostType, SID, JID) -> ReplacedPids when
       HostType :: mongooseim:host_type(),
+      SID :: sid(),
       JID :: jid:jid(),
       ReplacedPids :: [pid()].
-check_for_sessions_to_replace(HostType, JID) ->
+check_for_sessions_to_replace(HostType, {_, NewPid}, JID) ->
     #jid{luser = LUser, lserver = LServer, lresource = LResource} = JID,
     Sessions = ejabberd_sm_backend:get_sessions(LUser, LServer),
     %% TODO: Depending on how this is executed, there could be an unneeded
     %% replacement for max_sessions. We need to check this at some point.
     ReplacedRedundantSessions = check_existing_resources(LResource, Sessions),
     AllReplacedSessionPids = check_max_sessions(HostType, LUser, LServer, ReplacedRedundantSessions, Sessions),
-    [mongoose_c2s:exit(Pid, <<"Replaced by new connection">>) || Pid <- AllReplacedSessionPids],
+    [mongoose_c2s:exit(Pid, {replaced, NewPid}) || Pid <- AllReplacedSessionPids],
```

Comment:
is there an option when AllReplacedSessionPids list contains more than one item?

Reply:
I can't think of any, but I thought that there potentially might be some race condition. I decided not to change it here, but we can investigate.